programs/drift_vaults/src/instructions/deposit.rs

```diff
@@ -21,15 +23,44 @@ pub fn deposit<'info>(ctx: Context<'_, '_, '_, 'info, Deposit<'info>>, amount: u
     let mut vault = ctx.accounts.vault.load_mut()?;
     let mut vault_depositor = ctx.accounts.vault_depositor.load_mut()?;
 
-    // todo, use calculate_net_usd_value in margin.rs
-    let (net_usd_value, all_oracles_valid) = (100_u64, true);
+    let user = ctx.accounts.drift_user.load()?;
+    let spot_market_index = vault.spot_market_index;
+
+    let remaining_accounts_iter = &mut ctx.remaining_accounts.iter().peekable();
+    let AccountMaps {
+        perp_market_map,
+        spot_market_map,
+        mut oracle_map,
+    } = load_maps(
+        remaining_accounts_iter,
+        &MarketSet::new(),
+        &get_writable_perp_market_set(spot_market_index),
+        clock.slot,
+        None,
+    )?;
+
+    let (net_usd_value, all_oracles_valid) =
+        calculate_net_usd_value(&user, &perp_market_map, &spot_market_map, &mut oracle_map)?;
+
     validate!(all_oracles_valid, ErrorCode::Default)?;
-    vault_depositor.deposit(amount, net_usd_value, &mut vault, clock.unix_timestamp)?;
+    let non_negative_net_usd_value = net_usd_value.max(0).cast()?;
```

Comment:
should we just throw err if it is negative? also should maybe check that user isn't being liq'd?

Reply:
added